projects/pgai/pgai/vectorizer/features/features.py

```diff
@@ -1,24 +1,69 @@
 from functools import cached_property
+from typing import Literal, Self
```

Comment:
`Self` is only available in python 3.11, we went through this here: https://github.com/timescale/pgai/pull/573/files#diff-a36adde57fda86e43aca03bba8fd7f62e22b2e787e6f9b48028ba7b6a3fdde13

Reply:
is there anything blocking us from upgrading the library to a newer python version? The extension was limited by the python version built into plpython3u

Reply:
Upgrading the python version limits who can use the library in their own code. We should somewhat follow the supported bugfix schedule by python here imo: https://devguide.python.org/versions/